src/main/java/redis/clients/jedis/JedisClientSideCache.java

```diff
@@ -0,0 +1,45 @@
+package redis.clients.jedis;
+
+import redis.clients.jedis.exceptions.JedisException;
+
+public class JedisClientSideCache extends Jedis {
+
+  private final ClientSideCache cache;
+
+  public JedisClientSideCache(final HostAndPort hostPort, final JedisClientConfig config) {
+    this(hostPort, config, new ClientSideCache());
+  }
+
+  public JedisClientSideCache(final HostAndPort hostPort, final JedisClientConfig config,
+      ClientSideCache cache) {
+    super(hostPort, config);
+    if (config.getRedisProtocol() != RedisProtocol.RESP3) {
+      throw new JedisException("Client side caching is only supported with RESP3.");
+    }
+
+    this.cache = cache;
+    this.connection.setClientSideCache(cache);
+    clientTrackingOn();
+  }
+
+  private void clientTrackingOn() {
+    String reply = connection.executeCommand(new CommandObject<>(
+        new CommandArguments(Protocol.Command.CLIENT).add("TRACKING").add("ON").add("BCAST"),
```

Comment:
@chayim @uglide As I remember BCAST is not an option here

Reply:
Correct. @sazzad16